### PR TITLE
Added Koin integration

### PR DESCRIPTION
**Pull request by padobariya**
Added Koin integration

### CODE REVIEW

File: app/src/main/java/com/quipper/android/apollofrontpage/PostListFragment.kt [hunk 1]
```diff
@@ -5,21 +5,20 @@ import android.view.LayoutInflater
...
 class PostListFragment : Fragment(), PostListAdapter.PostListHandler {
 
     companion object {
-
         fun newInstance() = PostListFragment()
     }
 
     private lateinit var binding: PostListFragmentBinding
-    private val viewModel: PostListViewModel by viewModels()
+    private val viewModel: PostListViewModel by inject()
```
Comment (aruke) [inline app/src/main/java/com/quipper/android/apollofrontpage/PostListFragment.kt]: You can still use `by viewModel()` here.

File: app/build.gradle [hunk 2]
```diff
@@ -67,4 +67,10 @@ dependencies {
     // for apollo library
     compileOnly 'org.jetbrains:annotations:13.0'
     testCompileOnly 'org.jetbrains:annotations:13.0'
+
+    // Koin
+    implementation "org.koin:koin-android:${koinVersion}"
+    implementation "org.koin:koin-androidx-scope:${koinVersion}"
+    implementation "org.koin:koin-androidx-viewmodel:${koinVersion}"
+    implementation "org.koin:koin-androidx-fragment:${koinVersion}"
```
Comment (aruke) [inline app/build.gradle]: Better not use the libraries we are not using right now.

File: app/src/main/java/com/quipper/android/apollofrontpage/repository/impl/PostsRepositoryImpl.kt [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package com.quipper.android.apollofrontpage.repository.impl
+
+import com.quipper.android.apollofrontpage.model.Posts
+import com.quipper.android.apollofrontpage.repository.PostsRepository
+
+class PostsRepositoryImpl : PostsRepository {
```
Comment (aruke) [inline app/src/main/java/com/quipper/android/apollofrontpage/repository/impl/PostsRepositoryImpl.kt]: No need to add `Impl` suffix if it is already in `impl` packages.

File: app/src/main/java/com/quipper/android/apollofrontpage/PostListFragment.kt (hunk 1 above)
Comment (geckour) [inline app/src/main/java/com/quipper/android/apollofrontpage/PostListFragment.kt]: We should use `by viewModel()` as following the doc https://start.insert-koin.io/#/quickstart/android-viewmodel

> The by `viewModel()` function allows us to retrieve a ViewModel instance from Koin, linked to the Android ViewModelFactory.

File: app/src/main/java/com/quipper/android/apollofrontpage/app/ApolloFrontPageApp.kt [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+import org.koin.core.context.startKoin
+
+class ApolloFrontPageApp : Application() {
+    override fun onCreate() {
+        super.onCreate()
+        // Start Koin
+        startKoin {
+            androidLogger()
+            androidContext(this@ApolloFrontPageApp)
+            modules(repositoryModule)
```
Comment (geckour) [inline app/src/main/java/com/quipper/android/apollofrontpage/app/ApolloFrontPageApp.kt]: We should keep the PR simply so I think we should remove repository implementation from it

**Review by geckour — COMMENTED** [review]
📝 Commented

**Review by aruke — COMMENTED** [review]
Thanks for the code! Commented.

File: app/src/main/java/com/quipper/android/apollofrontpage/PostListFragment.kt (hunk 1 above)
Comment (padobariya) [reply app/src/main/java/com/quipper/android/apollofrontpage/PostListFragment.kt]: Resolved! I tried that earlier but somehow AS autocomplete was not showing that method. Seems we need to explicitly import that.

**Comment by padobariya** [comment]
@aruke @geckour Thanks for the review. Can you pls check again.

File: app/src/main/java/com/quipper/android/apollofrontpage/model/Posts.kt [hunk 5]
```diff
@@ -0,0 +1,5 @@
+package com.quipper.android.apollofrontpage.model
+
+data class Posts(var id: Int, var title: String, var votes: Int, var author: Author)
+
+data class Author(var firstName: String, var lastName: String)
```
Comment (oh-naoki) [inline app/src/main/java/com/quipper/android/apollofrontpage/model/Posts.kt:5]: [NITS] 
If these classes aren't used, we may remove :bow:

Comment (padobariya) [reply app/src/main/java/com/quipper/android/apollofrontpage/model/Posts.kt:5]: Sorry! It is already removed in the [next PR](https://github.com/quipper/apollo-frontpage-android-app/pull/6)

**Review by geckour — APPROVED** [review]
🍣

**Review by aruke — APPROVED** [review]
Looks good now!

**Pull request merged**